modules/openthread/platform/radio.c

```diff
@@ -385,6 +393,25 @@ void transmit_message(struct k_work *tx_job)
 	radio_api->set_channel(radio_dev, channel);
 	radio_api->set_txpower(radio_dev, get_transmit_power_for_channel(channel));
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+	if (sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset != 0) {
+		*(sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset) =
+			sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
+		uint8_t *time_ie =
+			sTransmitFrame.mPsdu + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeIeOffset;
+		uint64_t offset_plat_time =
+			otPlatTimeGet() + sTransmitFrame.mInfo.mTxInfo.mIeInfo->mNetworkTimeOffset;
+
+		*time_ie = sTransmitFrame.mInfo.mTxInfo.mIeInfo->mTimeSyncSeq;
+
+		*(++time_ie) = (uint8_t)(offset_plat_time & 0xff);
+		for (uint8_t i = 1; i < sizeof(uint64_t); i++) {
+			offset_plat_time = offset_plat_time >> 8;
+			*(++time_ie) = (uint8_t)(offset_plat_time & 0xff);
+		}
```

Comment:
Are we just writing 64-bit integer to the buffer in little endian here? We have `sys_put_le64()` helper in Zephyr (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/byteorder.h#L485).

Reply:
Changed for sys_put_le64. Thanks for the hint. Way cleaner!